### PR TITLE
[DOP-3137] Add tests workflow for core

### PR DESCRIPTION
**Pull request by dolfinus**
Added CI workflow to run unit tests for core components.

Tests are additionally marked with:
* corresponding connection type name, e.g. `s3`, `sftp` and so on
* `db_connection` - all connections which uses `spark` or `spark_mock` fixtures
* `file_connection` - all tests which import file connection classes (requires optional dependencies to be installed)
* `connection` = `db_connection` | `file_connection`

Core tests are ones NOT marked with `connection`.

Also:
* test `test_secure_str_and_repr` was removed, each DB connection is tested separately
* Some `hwm_store` integration tests now are unit tests

### CODE REVIEW

File: tests/tests_unit/tests_hwm_store_unit/test_yaml_hwm_store_unit.py [hunk 1]
```diff
@@ -8,6 +8,69 @@
 from onetl.hwm.store import HWMStoreManager, YAMLHWMStore
 
 
+def test_hwm_store_yaml_path(request, tmp_path_factory, hwm_delta):
+    hwm, _delta = hwm_delta
+    folder: Path = tmp_path_factory.mktemp("someconf")
```
Comment (andy-takker) [inline tests/tests_unit/tests_hwm_store_unit/test_yaml_hwm_store_unit.py:13]:
```suggestion
    folder: Path = tmp_path_factory.mktemp("someconf") # noqa: S306
```

Comment (dolfinus) [reply tests/tests_unit/tests_hwm_store_unit/test_yaml_hwm_store_unit.py:13]: Почему?

Comment (andy-takker) [reply tests/tests_unit/tests_hwm_store_unit/test_yaml_hwm_store_unit.py:13]: разве он не должен триггерить ? https://bandit.readthedocs.io/en/latest/blacklists/blacklist_calls.html#b306-mktemp-q

Comment (dolfinus) [reply tests/tests_unit/tests_hwm_store_unit/test_yaml_hwm_store_unit.py:13]: Это не модуль, а фикстура pytest

**Review by dmitry-pedchenko — APPROVED** [review]

**Review by andy-takker — APPROVED** [review]

**Pull request merged**